### PR TITLE
feat(k8s): add vrack support

### PR DESCRIPTION
**Pull request by d33d33**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-check#240.0 ✘  
* 
  * checks ✘

**Comment by ovh-cds** [comment]
CDS Report terraform-provider-ovh-testacc#240.0 ✘  
* 
  * checks ✘

File: ovh/data_source_ovh_cloud_project_kube.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ func dataSourceCloudProjectKube() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"privateNetworkId": {
```
Comment (yanndegat) [inline ovh/data_source_ovh_cloud_project_kube.go:33]: should be named private_network_id

File: ovh/data_source_ovh_cloud_project_kube.go [hunk 2]
```diff
@@ -93,6 +97,7 @@ func dataSourceCloudProjectKubeRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("name", res.Name)
 	d.Set("next_upgrade_versions", res.NextUpgradeVersions)
 	d.Set("nodes_url", res.NodesUrl)
+	d.Set("privateNetworkId", res.PrivateNetworkId)
```
Comment (yanndegat) [inline ovh/data_source_ovh_cloud_project_kube.go:100]: should be named private_network_id

File: ovh/resource_ovh_cloud_project_kube.go [hunk 3]
```diff
@@ -43,6 +43,11 @@ func resourceCloudProjectKube() *schema.Resource {
 				Optional: true,
 				ForceNew: true,
 			},
+			"privateNetworkId": {
```
Comment (yanndegat) [inline ovh/resource_ovh_cloud_project_kube.go:46]: should be named private_network_id

File: ovh/data_source_ovh_cloud_project_kube.go [hunk 4]
```diff
@@ -30,6 +30,10 @@ func dataSourceCloudProjectKube() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"privateNetworkId": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (yanndegat) [inline ovh/data_source_ovh_cloud_project_kube.go:35]: should be computed only

File: ovh/resource_ovh_cloud_project_kube.go [hunk 5]
```diff
@@ -43,6 +43,11 @@ func resourceCloudProjectKube() *schema.Resource {
...
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```
Comment (yanndegat) [inline ovh/resource_ovh_cloud_project_kube.go:49]: are you sure it's forcenew ? (if there's a change, it should destroy/create the cluster ?)

File: website/docs/r/cloud_project_kube.html.markdown [hunk 6]
```diff
@@ -44,6 +44,7 @@ The following attributes are exported:
 * `name` - See Argument Reference above.
 * `region` - See Argument Reference above.
 * `version` - See Argument Reference above.
+* `privateNetworkId` - OpenStack private network (or vrack) ID to use.
```
Comment (yanndegat) [inline website/docs/r/cloud_project_kube.html.markdown:47]: should be named private_network_id

File: website/docs/d/cloud_project_kube.html.markdown [hunk 7]
```diff
@@ -37,6 +37,7 @@ The following attributes are exported:
 * `name` - The name of the managed kubernetes cluster.
 * `region` - The OVH public cloud region ID of the managed kubernetes cluster.
 * `version` - Kubernetes version of the managed kubernetes cluster.
+* `privateNetworkId` - OpenStack private network (or vrack) ID to use.
```
Comment (yanndegat) [inline website/docs/d/cloud_project_kube.html.markdown:40]: should be named private_network_id

File: ovh/resource_ovh_cloud_project_kube.go (hunk 5 above)
Comment (mhurtrel) [reply ovh/resource_ovh_cloud_project_kube.go:49]: Yes, can't be changed afterwards

File: ovh/data_source_ovh_cloud_project_kube.go (hunk 2 above)
Comment (yanndegat) [reply ovh/data_source_ovh_cloud_project_kube.go]: i've double- checked it, i think you forgot to pass the network id to the POST api call, i'd like to release the provider this morning so i think i'll add it for you if you dont mind

**Pull request merged**